tools/system_libs.py

```diff
@@ -147,7 +147,15 @@ def get_wasm_libc_rt_files():
     path_components=['system', 'lib', 'libc'],
     filenames=['emscripten_memcpy.c', 'emscripten_memset.c',
                'emscripten_memmove.c'])
-  return math_files + other_files
+  # Calls to iprintf can be generated during codegen
+  iprintf_files = files_in_path(
+    path_components=['system', 'lib', 'libc', 'musl', 'src', 'stdio'],
+    filenames=['__towrite.c', '__overflow.c', 'fwrite.c', 'fputs.c',
+               'printf.c', 'puts.c'])
+  iprintf_files += files_in_path(
+    path_components=['system', 'lib', 'libc', 'musl', 'src', 'string'],
+    filenames=['strlen.c'])
+  return math_files + other_files + iprintf_files
```

Comment:
these files are both in libc_rt here and in libc - is there no problem with that? I guess it's fine since it's not linked in a second time, it's only pulled in as needed? (if so, could all of libc be in libc_rt?)

Reply:
They are included in libc_rt_wasm explicitly, but *excluded* from libc (added to the blacklist).

Reply:
Thanks!